pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -169,8 +176,10 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 		return errors.Wrap(err, "failed to get cluster provider config")
 	}
 
+	// status.InstanceID is nil, so don't do this
 	if status.InstanceID == nil {
```

Comment:
Would it make sense to try and lookup by tags here? Currently we could potentially orphan an instance if create is called, the instance is successfully created, the object update fails, delete is processed before the requeued create is processed.

Reply:
yes, I agree. can that be a followup issue or should it be part of this PR?